### ISSUE TITLE
Why rescale the gradient with batch size?

### ISSUE THREAD

**Issue by huaxinxiao** [apache/incubator-mxnet#4761]
I am training a network for semantic segmentation by using the following softmax symbol and setting:
`softmax  = mx.symbol.SoftmaxOutput(data=upscore, label=label, multi_output=True, normalization='valid', use_ignore=True, ignore_label=255, name='softmax')`

`model = mx.model.FeedForward(
    ctx                = devs,
    symbol             = net_symbol,
    num_epoch          = args.num_epochs,
    learning_rate      = args.lr,
    momentum           = 0.9,
    wd                 = 0.0001, 
    optimizer          = 'nag',
    initializer        = mx.init.MSRAPrelu(),
    **model_args)`
I found that the  default value of` rescale_grad`  is setting to `rescale_grad=(1.0/batch_size)` in [model.py](https://github.com/dmlc/mxnet/blob/master/python/mxnet/model.py#L794)

I am confused about the setting of rescale_grad. Because I  think the gradient in the [softmax output](https://github.com/dmlc/mxnet/blob/master/src/operator/softmax_output-inl.h#L175) has been normalized properly by `valid_cnt`.  Could someone explain the `rescale_grad` setting?

Thanks!

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
